From 285a6a30d7100729c724831c10ec3db9238f942e Mon Sep 17 00:00:00 2001
From: Omar Eissa <oeissa@google.com>
Date: Wed, 27 Aug 2025 13:10:46 +0000
Subject: [PATCH] Allow resetting the voice recognition service if its package
 uninstalled

Previously, if voice interaction service is set on the device, and the
current voice recognition service package gets uninstalled, we were
returning early from onSomePackagesChanged preventing the logic of
resetting the current recognizer from being invoked.

Bug: 352024705
Test: Manual, uninstall voice recognition app and check adb shell settings get secure voice_recognition_service
Flag: EXEMPT CVE_FIX
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:e8d07dd802a9172e68a9a043804fe11c48180e7f)
Merged-In: I563ad50dac05a019c2ce8b727b5e5744ba848cc0
Change-Id: I563ad50dac05a019c2ce8b727b5e5744ba848cc0
---
 .../voiceinteraction/VoiceInteractionManagerService.java     | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/services/voiceinteraction/java/com/android/server/voiceinteraction/VoiceInteractionManagerService.java b/services/voiceinteraction/java/com/android/server/voiceinteraction/VoiceInteractionManagerService.java
index 605af03f112f..b3fa0ea029ef 100644
--- a/services/voiceinteraction/java/com/android/server/voiceinteraction/VoiceInteractionManagerService.java
+++ b/services/voiceinteraction/java/com/android/server/voiceinteraction/VoiceInteractionManagerService.java
@@ -2457,10 +2457,11 @@ public class VoiceInteractionManagerService extends SystemService {
                                 switchImplementationIfNeededLocked(true);
                             }
                         }
-                        return;
                     }
 
-                    if (curAssistant != null) {
+                    // If interactor isn't null, then we would have done the needed checks already
+                    // in the above code.
+                    if (curInteractor == null && curAssistant != null) {
                         int change = isPackageDisappearing(curAssistant.getPackageName());
                         if (change == PACKAGE_PERMANENT_CHANGE) {
                             // If the currently set assistant is being removed, then we should
-- 
2.52.0.rc1.455.g30608eb744-goog

